### ISSUE TITLE
Package Control won't install in linux for sublime 3

### ISSUE THREAD

**Issue by imagineer64** [wbond/package_control#1100]
I am seeing errors in the console when running the package control install code. The entire console output is below:

libgio version is less than 2.26, single instance support disabled
startup, version: 3083 linux x64 channel: stable
executable: /home/sublime/sublime_text_3/sublime_text
working dir: /
packages path: /home/.config/sublime-text-3/Packages
state path: /home/.config/sublime-text-3/Local
zip path: /home/sublime/sublime_text_3/Packages
zip path: /home/.config/sublime-text-3/Installed Packages
ignored_packages: ["Vintage"]
pre session restore time: 0.163053
startup time: 0.478633
first paint time: 0.494123
reloading plugin Default.block
reloading plugin Default.comment
reloading plugin Default.copy_path
reloading plugin Default.delete_word
reloading plugin Default.detect_indentation
reloading plugin Default.duplicate_line
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.indentation
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_file_settings
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.quick_panel
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.swap_line
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.trim_trailing_white_space
reloading plugin CSS.css_completions
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
plugins loaded
import urllib.request,os,hashlib; h = '2915d1851351e5ee549c20394736b442' + '8bc59f460fa1548d1514676163dafc88'; pf = 'Package Control.sublime-package'; ipp = sublime.installed_packages_path(); urllib.request.install_opener( urllib.request.build_opener( urllib.request.ProxyHandler()) ); by = urllib.request.urlopen( 'http://packagecontrol.io/' + pf.replace(' ', '%20')).read(); dh = hashlib.sha256(by).hexdigest(); print('Error validating download (got %s instead of %s), please try manual install' % (dh, h)) if dh != h else open(os.path.join( ipp, pf), 'wb' ).write(by)
Traceback (most recent call last):
  File "./urllib/request.py", line 1248, in do_open
  File "./http/client.py", line 1065, in request
  File "./http/client.py", line 1103, in _send_request
  File "./http/client.py", line 1061, in endheaders
  File "./http/client.py", line 906, in _send_output
  File "./http/client.py", line 844, in send
  File "./http/client.py", line 822, in connect
  File "./socket.py", line 435, in create_connection
  File "./socket.py", line 426, in create_connection
TimeoutError: [Errno 110] Connection timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "./urllib/request.py", line 156, in urlopen
  File "./urllib/request.py", line 469, in open
  File "./urllib/request.py", line 487, in _open
  File "./urllib/request.py", line 447, in _call_chain
  File "./urllib/request.py", line 1268, in http_open
  File "./urllib/request.py", line 1251, in do_open
urllib.error.URLError: <urlopen error [Errno 110] Connection timed out>
Unable to fetch update url contents

Is this a problem with the proxy at my company or version of python. It seems that sublime comes packaged with it's own python and shouldn't have any dependency on my linux system's python. I was able to successfully install package control for my windows machine that I use at the same company so it seems to connect using the proxy.

Thanks

**Comment by imagineer64** [comment]
Ok, I decided to start from scratch since I had just started to setup sublime 3. I reinstalled build 3103 and I changed the global settings to this for my company proxy. I obfuscated the proxy name:

```
{
  "http_proxy": "proxy.xyz.com:911",
  "https_proxy": "proxy.xyz.com:912"
}
```
I then ran the python commands to install package control from https://packagecontrol.io/installation#st3

The console output is below:

import urllib.request,os,hashlib; h = '2915d1851351e5ee549c20394736b442' + '8bc59f460fa1548d1514676163dafc88'; pf = 'Package Control.sublime-package'; ipp = sublime.installed_packages_path(); urllib.request.install_opener( urllib.request.build_opener( urllib.request.ProxyHandler()) ); by = urllib.request.urlopen( 'http://packagecontrol.io/' + pf.replace(' ', '%20')).read(); dh = hashlib.sha256(by).hexdigest(); print('Error validating download (got %s instead of %s), please try manual install' % (dh, h)) if dh != h else open(os.path.join( ipp, pf), 'wb' ).write(by)
186789
reloading plugin Package Control.1_reloader
reloading plugin Package Control.2_bootstrap
reloading plugin Package Control.Package Control
Package Control: Not running package cleanup since bootstrapping is not yet complete
reloading Packages/Package Control/Package Control.sublime-settings
Package Control: Downloading bootstrapped dependency ssl-linux
Package Control: Successfully downloaded bootstraped dependency ssl-linux
reloading plugin 0_package_control_loader.00-package_control
Exception in thread Thread-1:
Traceback (most recent call last):
  File "./threading.py", line 901, in _bootstrap_inner
  File "./threading.py", line 858, in run
  File "package_control.bootstrap in /home/.config/sublime-text-3/Installed Packages/Package Control.sublime-package", line 193, in bootstrap_dependency
  File "package_control.loader in /home/.config/sublime-text-3/Installed Packages/Package Control.sublime-package", line 207, in add
  File "/home/.config/sublime-text-3/Installed Packages/0_package_control_loader.sublime-package/01-ssl-linux.py", line 6, in <module>
ImportError: No module named 'package_control'

I then added the following package control settings and restarted sublime:

```
{
   "bootstrapped": true,
   "http_proxy": "proxy.xyz.com:911",
   "https_proxy": "proxy.xyz.com:912",
   "debug":true
}
```

libgio version is less than 2.26, single instance support disabled
startup, version: 3103 linux x64 channel: stable
executable: /home/sublime/sublime_text_3/sublime_text
working dir: /
packages path: /home/.config/sublime-text-3/Packages
state path: /home/.config/sublime-text-3/Local
zip path: /home/sublime/sublime_text_3/Packages
zip path: /home/.config/sublime-text-3/Installed Packages
ignored_packages: ["Vintage"]
pre session restore time: 0.176711
startup time: 0.414337
first paint time: 0.52517
reloading plugin Default.block
reloading plugin Default.comment
reloading plugin Default.convert_syntax
reloading plugin Default.copy_path
reloading plugin Default.delete_word
reloading plugin Default.detect_indentation
reloading plugin Default.duplicate_line
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.indentation
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_file_settings
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.profile
reloading plugin Default.profile_syntax_definition
reloading plugin Default.quick_panel
reloading plugin Default.run_syntax_tests
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.swap_line
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.trim_trailing_white_space
reloading plugin CSS.css_completions
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin Python.syntax_test_python
reloading plugin 0_package_control_loader.00-package_control
reloading plugin 0_package_control_loader.01-ssl-linux
Package Control: Linux SSL: _ssl module import error - libssl.so.1.0.0: cannot open shared object file: No such file or directory
Package Control: Linux SSL: _ssl module import error - libssl.so.10: cannot open shared object file: No such file or directory
Package Control: Linux SSL: successfully loaded _ssl module for libssl.so.0.9.8
reloading plugin Package Control.1_reloader
reloading plugin Package Control.2_bootstrap
reloading plugin Package Control.Package Control
plugins loaded
Package Control: Download Debug
  URL: https://packagecontrol.io/submit?operation=install&version=3.1.2&package=Package+Control&package_control_version=3.1.2&sublime_version=3103&sublime_platform=linux
  Timeout: 30
  Resolved IP: 50.116.33.29
  Resolved IPv6: 2600:3c02::f03c:91ff:fedf:6883
Package Control: Urllib Debug Proxy
  http_proxy: proxy.xyz.com:911
  https_proxy: proxy.xyz.com:912
  proxy_username: 
  proxy_password: 
Package Control: Created blank user CA bundle
Package Control: Regenerated the merged CA bundle from the system and user CA bundles
Package Control: Urllib HTTPS Debug General
  Connecting to proxy.xyz.com on port 912
Package Control: Urllib HTTPS Debug Write
  CONNECT packagecontrol.io:443 HTTP/1.1
  Host: packagecontrol.io:443
  Proxy-Connection: Keep-Alive
  User-Agent: Package Control v3.1.2
Package Control: Urllib HTTPS Debug Read
  HTTP/1.1 200 Connection established
Package Control: Urllib HTTPS Debug General
  Upgrading connection to SSL using CA certs file at /home/.config/sublime-text-3/Packages/User/Package Control.merged-ca-bundle
Package Control: Error submitting usage information. URL error unknown error (_ssl.c:2228) downloading https://packagecontrol.io/submit?operation=install&version=3.1.2&package=Package+Control&package_control_version=3.1.2&sublime_version=3103&sublime_platform=linux.
Package Control: Installing 1 missing dependency
Package Control: Fetching list of available packages
  Platform: linux-x64
  Sublime Text Version: 3103
  Package Control Version: 3.1.2
Package Control: Download Debug
  URL: https://packagecontrol.io/channel_v3.json
  Timeout: 30
  Resolved IP: 50.116.33.29
  Resolved IPv6: 2600:3c02::f03c:91ff:fedf:6883
Package Control: Urllib HTTPS Debug General
  Connecting to proxy.xyz.com on port 912
Package Control: Urllib HTTPS Debug Write
  CONNECT packagecontrol.io:443 HTTP/1.1
  Host: packagecontrol.io:443
  Proxy-Connection: Keep-Alive
  User-Agent: Package Control v3.1.2
Package Control: Urllib HTTPS Debug Read
  HTTP/1.1 200 Connection established
Package Control: Urllib HTTPS Debug General
  Upgrading connection to SSL using CA certs file at /home/.config/sublime-text-3/Packages/User/Package Control.merged-ca-bundle
Package Control: Error downloading channel. URL error unknown error (_ssl.c:2228) downloading https://packagecontrol.io/channel_v3.json.
Package Control: The dependency specified, bz2, is not available
Package Control: Fetching list of available packages
  Platform: linux-x64
  Sublime Text Version: 3103
  Package Control Version: 3.1.2
Package Control: Download Debug
  URL: https://packagecontrol.io/channel_v3.json
  Timeout: 30
  Resolved IP: 50.116.33.29
  Resolved IPv6: 2600:3c02::f03c:91ff:fedf:6883
Package Control: Urllib HTTPS Debug General
  Connecting to proxy.xyz.com on port 912
Package Control: Urllib HTTPS Debug Write
  CONNECT packagecontrol.io:443 HTTP/1.1
  Host: packagecontrol.io:443
  Proxy-Connection: Keep-Alive
  User-Agent: Package Control v3.1.2
Package Control: Urllib HTTPS Debug Read
  HTTP/1.1 200 Connection established
Package Control: Urllib HTTPS Debug General
  Upgrading connection to SSL using CA certs file at /home/.config/sublime-text-3/Packages/User/Package Control.merged-ca-bundle
Package Control: Error downloading channel. URL error unknown error (_ssl.c:2228) downloading https://packagecontrol.io/channel_v3.json.
Package Control: Fetching list of available packages
  Platform: linux-x64
  Sublime Text Version: 3103
  Package Control Version: 3.1.2
Package Control: Download Debug
  URL: https://packagecontrol.io/channel_v3.json
  Timeout: 30
  Resolved IP: 50.116.33.29
  Resolved IPv6: 2600:3c02::f03c:91ff:fedf:6883
Package Control: Urllib HTTPS Debug General
  Connecting to proxy.xyz.com on port 912
Package Control: Urllib HTTPS Debug Write
  CONNECT packagecontrol.io:443 HTTP/1.1
  Host: packagecontrol.io:443
  Proxy-Connection: Keep-Alive
  User-Agent: Package Control v3.1.2
Package Control: Urllib HTTPS Debug Read
  HTTP/1.1 200 Connection established
Package Control: Urllib HTTPS Debug General
  Upgrading connection to SSL using CA certs file at /home/.config/sublime-text-3/Packages/User/Package Control.merged-ca-bundle
Package Control: Error downloading channel. URL error unknown error (_ssl.c:2228) downloading https://packagecontrol.io/channel_v3.json.
Package Control: No updated packages
Unable to fetch update url contents

Unfortunately running install packages gives the message that there are no packages available for installation.

Any help would be appreciated.

**Comment by imagineer64** [comment]
I have resolved this issue by using the following steps.

I used the following global settings (my company requires a proxy):

```
{
  "http_proxy": "proxy.xyz.com:911"
}
```

I also used the following package control settings. I had to use curl as the default downloader since I couldn't determine why urllib didn't work.

```
{
	"bootstrapped": true,
	"downloader_precedence":
	{
		"linux": [ "curl", "urllib",	"wget" ],
		"osx": [ "urllib" ],
		"windows": [ "wininet" ]
	},
	"http_proxy": "proxy-us.intel.com:911"
}
```

Finally, I had to make sure that curl worked correctly from the unix prompt before starting sublime. It turns out there was another path in my $LD_LIBRARY_PATH which was referencing a lib that was incompatible with my curl. I was getting an error when running curl so for simplicity I just created a unix alias for sublime that did a unsetenv LD_LIBRARY_PATH because I didn't need the paths for sublime only other tools.
